### PR TITLE
Fix psh_i test case on Ubuntu 18.04

### PR DESCRIPTION
**Pull request by neo954**
Fix github issue #5466.

### CODE REVIEW

**Comment by neo954** [comment]
The unit test result
```
# xcattest -f xcat-test.conf -t psh_i
xCAT automated test started at Mon Aug  6 00:15:52 2018
******************************
loading Configure file
******************************
TABLE:site
    key = nameservers
    value = 10.0.0.103
    key = master
    value = 10.3.17.3
    key = domain
    value = pok.stglabs.ibm.com
    key = forwarders
    value = 10.0.0.103
    key = consoleondemand
    value = yes
TABLE:passwd
    key = system
    username = root
    password = cluster
Varible:
    GATEWAY = 10.0.0.103
    CN = c910f03c17k04
    UBUNTU_MIGRATION2_VERSION = 2.13.4
    ISO = /media/xcat/iso/redhat/7.4/RHEL-7.4-20170711.0-Server-ppc64le-dvd1.iso
    SN = c910f03c17k04
    bmcrange = 10.4.30.254
    OS = rhels
    MIGRATION1_DEP = http://xcat.org/files/xcat/xcat-dep/2.x_Linux/xcat-dep-2.12.4-linux.tar.bz2
    bmcpasswd = PASSW0RD
    NETDRIVER = ibmveth
    MIGRATION2_VERSION = 2.13.4
    LATEST_VERSION = 2.13.5
    MIGRATION1_CORE = http://xcat.org/files/xcat/xcat-core/2.12.x_Linux/xcat-core/xcat-core-2.12.4-linux.tar.bz2
    MIGRATION2_DEP = http://xcat.org/files/xcat/xcat-dep/2.x_Linux/xcat-dep-2.13.4-linux.tar.bz2
    UBUNTU_MIGRATION1_DEP = http://xcat.org/files/xcat/xcat-dep/2.x_Ubuntu/xcat-dep-2.12.4-ubuntu.tar.bz2
    bmcusername = USERID
    MN = c910f03c17k03
    MIGRATION2_CORE = http://xcat.org/files/xcat/xcat-core/2.13.x_Linux/xcat-core/xcat-core-2.13.4-linux.tar.bz2
    extntpserversip = 10.0.0.103
    SECONDNIC = eth4
    INSTALLNICIP = 10.3.1.9
    NODEIP = 10.3.1.9
    NETBOOTDIR = /opt/xcat/share/xcat/netboot/rh
    MaxLparID = 32
    range = 10.4-5.24-25.1
    STATELESS = rhels7.2-ppc64-netboot-compute
    MIGRATION1_VERSION = 2.12.4
    XCAT_DATABASE = MySQL
    THIRDNIC = eth5
    UBUNTU_MIGRATION2_DEP = http://xcat.org/files/xcat/xcat-dep/2.x_Ubuntu/xcat-dep-2.13.4-ubuntu.tar.bz2
    UBUNTU_MIGRATION1_CORE = http://xcat.org/files/xcat/xcat-core/2.12.x_Ubuntu/xcat-core/xcat-core-2.12.4-ubuntu.tar.bz2
    UBUNTU_MIGRATION1_VERSION = 2.12.4
    STATELITE = rhels7.2-ppc64-statelite-compute
    UBUNTU_MIGRATION2_CORE = http://xcat.org/files/xcat/xcat-core/2.13.x_Ubuntu/xcat-core/xcat-core-2.13.4-ubuntu.tar.bz2
    DATE = 2016
    MINIISO = /mini.iso
******************************
Initialize xCAT test environment by definition in configure file
******************************
chtab key=nameservers site.value=10.0.0.103
chtab key=master site.value=10.3.17.3
chtab key=domain site.value=pok.stglabs.ibm.com
chtab key=forwarders site.value=10.0.0.103
chtab key=consoleondemand site.value=yes
chtab key=system passwd.username=root passwd.password=cluster
******************************
To detect current test environment
******************************
Detecting: ARCH = ppc64le
Detecting: HCP = kvm
******************************
loading test cases
******************************
To run:
psh_i
******************************
Start to run test cases
******************************
------START::psh_i::Time:Mon Aug  6 00:15:54 2018------

RUN:cp /etc/hosts /etc/hostsBK [Mon Aug  6 00:15:54 2018]
ElapsedTime:0 sec
RETURN rc = 0
OUTPUT:
CHECK:rc == 0	[Pass]

RUN:interface=`xdsh c910f03c17k04 'ip route' | awk '/default/ { for (i = 1; i <= NF; ++i) if ("dev" == $i) { print $(i + 1) ; exit } }'`;sed -i "/c910f03c17k04/s/$/ c910f03c17k04-$interface/g" '/etc/hosts';psh -i $interface c910f03c17k04 'uptime' [Mon Aug  6 00:15:54 2018]
ElapsedTime:2 sec
RETURN rc = 0
OUTPUT:
c910f03c17k04-enp0s1: Warning: Permanently added 'c910f03c17k04-enp0s1' (ECDSA) to the list of known hosts.
c910f03c17k04-enp0s1:  00:15:56 up 2 days, 20:14,  1 user,  load average: 0.00, 0.00, 0.00
CHECK:output =~ c910f03c17k04	[Pass]
CHECK:rc == 0	[Pass]

RUN:rm -rf /etc/hosts [Mon Aug  6 00:15:56 2018]
ElapsedTime:0 sec
RETURN rc = 0
OUTPUT:

RUN:cp /etc/hostsBK /etc/hosts [Mon Aug  6 00:15:56 2018]
ElapsedTime:0 sec
RETURN rc = 0
OUTPUT:
CHECK:rc == 0	[Pass]

------END::psh_i::Passed::Time:Mon Aug  6 00:15:56 2018 ::Duration::2 sec------
------Total: 1 , Failed: 0------

xCAT automated test finished at Mon Aug  6 00:15:56 2018
Please check results in the /opt/xcat/bin/../share/xcat/tools/autotest/result/,
and see /opt/xcat/bin/../share/xcat/tools/autotest/result//failedcases.20180806001552 file for failed cases.
see /opt/xcat/bin/../share/xcat/tools/autotest/result//performance.report.20180806001552 file for time consumption
```

**Comment by neo954** [comment]
On RHEL 7, RHEL 8, Ubuntu 16.04
```
# ip route | awk '/default/ { print }'
default via 10.0.0.103 dev eth0
```

On Ubuntu 18.04 static IPv4
```
# ip route | awk '/default/ { print }'
default via 10.0.0.103 dev eth2 onlink
```

On Ubuntu 18.04 DHCP client
```
# ip route | awk '/default/ { print }'
default via 10.0.0.103 dev enp0s1 proto dhcp src 10.3.17.4 metric 100
```

File: xCAT-test/autotest/testcase/psh/cases0 [hunk 1]
```diff
@@ -49,7 +49,7 @@ description: psh -i interface CN 'uptime' in linux
 os:Linux
 cmd:cp /etc/hosts /etc/hostsBK
 check:rc==0
-cmd:interface=`xdsh $$CN 'ip route' | awk '/default/ { print $NF }'`;sed -i "/$$CN/s/$/ $$CN-$interface/g" '/etc/hosts';psh -i $interface $$CN 'uptime'
+cmd:interface=`xdsh $$CN 'ip route' | awk '/default/ { for (i = 1; i <= NF; ++i) if ("dev" == $i) { print $(i + 1) ; exit } }'`;sed -i "/$$CN/s/$/ $$CN-$interface/g" '/etc/hosts';psh -i $interface $$CN 'uptime'
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/psh/cases0]: It is better to add below line **before** and **after** line 52.
```
cmd: grep "$$CN"  /etc/hosts
```
Due to the failed cases show below error:
```
RUN:interface=`xdsh f6u13k18 'ip route' | awk '/default/ { print $NF }'`;sed -i "/f6u13k18/s/$/ f6u13k18-$interface/g" '/etc/hosts';psh -i $interface f6u13k18 'uptime' [Sun Aug  5 08:03:31 2018]
ElapsedTime:1 sec
RETURN rc = 0
OUTPUT:
Warning: Invalid nodes in noderange:node100
CHECK:output =~ f6u13k18	[Failed]
```
Not sure why it get ``Warning: Invalid nodes in noderange:node100``.

**Comment by neo954** [comment]
Thanks, @hu-weihua

**Pull request merged**